src/main/java/com/redhat/devtools/lsp4ij/features/formatting/LSPClientSideOnTypeFormattingTypedHandler.java

```diff
@@ -0,0 +1,292 @@
+                    String formatOnCloseBraceCharacters = formattingFeature.getFormatOnCloseBraceCharacters(file);
+                    Character openBraceChar = bracePair.getKey();
+                    Character closeBraceChar = bracePair.getValue();
+                    if (StringUtil.isEmpty(formatOnCloseBraceCharacters) ||
+                        (formatOnCloseBraceCharacters.indexOf(closeBraceChar) > -1)) {
+                        return handleCloseBraceTyped(
+                                project,
+                                editor,
+                                file,
+                                formattingFeature,
+                                openBraceChar,
+                                closeBraceChar
+                        );
+                    }
+                }
+            }
+
+            // Statement terminators
+            if (formattingFeature.isFormatOnStatementTerminator(file)) {
+                String formatOnStatementTerminatorCharacters = formattingFeature.getFormatOnStatementTerminatorCharacters(file);
+                if (StringUtil.isNotEmpty(formatOnStatementTerminatorCharacters) &&
+                    (formatOnStatementTerminatorCharacters.indexOf(charTyped) > -1)) {
+                    return handleStatementTerminatorTyped(
+                            project,
+                            editor,
+                            file,
+                            formattingFeature,
+                            charTyped
+                    );
+                }
+            }
+
+            // Completion triggers
+            if (formattingFeature.isFormatOnCompletionTrigger(file) &&
+                // It must be a completion trigger character for the language no matter what
+                LSPCompletionTriggerTypedHandler.hasLanguageServerSupportingCompletionTriggerCharacters(charTyped, project, file)) {
+                // But the subset that should trigger completion can be configured
+                String formatOnCompletionTriggerCharacters = formattingFeature.getFormatOnCompletionTriggerCharacters(file);
+                if (StringUtil.isEmpty(formatOnCompletionTriggerCharacters) ||
+                    (formatOnCompletionTriggerCharacters.indexOf(charTyped) > -1)) {
+                    return handleCompletionTriggerTyped(
+                            project,
+                            editor,
+                            file
+                    );
+                }
+            }
+        }
+
+        return super.charTyped(charTyped, project, editor, file);
+    }
+
+    @Nullable
+    private static LSPFormattingFeature getClientConfigurationSettings(@NotNull PsiFile file) {
+        Project project = file.getProject();
+        // Client-side on-type formatting shouldn't trigger a language server to start
+        Set<LanguageServerWrapper> startedLanguageServers = LanguageServiceAccessor.getInstance(project).getStartedServers();
```

Comment:
Please remove this code and uses getLanguageServers with proper filter (ex  started status) like you did in an another pr

Reply:
Done. The same commit also ensures that it doesn't even attempt range formatting if not supported by the language server, and I added a Go language unit test to confirm that change since that language server supports file formatting but not range formatting.